src/server/Connection.ts

```diff
@@ -217,13 +217,9 @@ export default class Connection extends EventEmitter<{
 
       const packetID = payload.readVarInt();
       this.emit("message", { packetID, payload });
-    }
+    } while (this.buffer.remaining() > 0);
 
-    if (this.buffer.remaining() > 0) {
```

Comment:
Forgot to mention that this condition is totally useless as if there is some incomplete data, you will most likely get an index out-of-bounds error generated from the while loop instead. :raised_hands: